From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Sat, 10 Feb 2018 15:01:27 -0500
Subject: tweak color computation looking to the unity launcher side

UbuntuSpecific: tweak for unity, nothing upstream uses the color computation
yet anyway so it's not an issue. Thanks gord.
---
 libgnome-desktop/gnome-bg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libgnome-desktop/gnome-bg.c b/libgnome-desktop/gnome-bg.c
index 7985f82..3f69d65 100644
--- a/libgnome-desktop/gnome-bg.c
+++ b/libgnome-desktop/gnome-bg.c
@@ -2434,7 +2434,7 @@ pixbuf_average_value (GdkPixbuf *pixbuf,
 	if (gdk_pixbuf_get_has_alpha (pixbuf)) {
 		for (row = 0; row < height; row++) {
 			p = pixels + (row * row_stride);
-			for (column = 0; column < width; column++) {
+			for (column = 0; column < width / 2; column++) {
 				r = *p++;
 				g = *p++;
 				b = *p++;
@@ -2451,7 +2451,7 @@ pixbuf_average_value (GdkPixbuf *pixbuf,
 	} else {
 		for (row = 0; row < height; row++) {
 			p = pixels + (row * row_stride);
-			for (column = 0; column < width; column++) {
+			for (column = 0; column < width / 2; column++) {
 				r = *p++;
 				g = *p++;
 				b = *p++;
